src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
```

Comment:
PullProofAI

Consider adding input validation to ensure `request.body` is present and valid before stringifying it.
```diff
+     if (!request.body || typeof request.body !== 'object') {
+         throw new BadRequestException('Invalid request body');
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->